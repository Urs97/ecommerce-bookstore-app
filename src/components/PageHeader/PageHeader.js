import React from 'react';
import './PageHeader.scss';

import { Element as ScrollElement } from 'react-scroll';

const PageHeader = ({ title, color }) => {
    return (
        <>
            <section className="page-header-nav-container" />
            <ScrollElement 
                name="page-header" 
                className={`page-header-category-title page-header-bg-${color}`}
                >
                <h1 className="title">{title}</h1>
            </ScrollElement>
        </>
    )
};

export default PageHeader;
