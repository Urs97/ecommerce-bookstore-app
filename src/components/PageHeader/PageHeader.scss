@import '../../styles/shared.scss';

.page-header-nav-container {
    width: 100%;
    height: 12em;
    background-color: $grey;
}

.page-header-category-title {
    width: 100%;
    height: 15em;
    background-color: $dark-blue;
    display: flex;
    justify-content: center;
    align-items: center;

    h1 {
        font-size: 3.5em;
        color: #FFFFFF;
    }
}

.page-header-bg-blue {
    background-color: $dark-blue;
}

.page-header-bg-green {
    background-color: $green;
}

.page-header-bg-yellow {
    background-color: $yellow;
}