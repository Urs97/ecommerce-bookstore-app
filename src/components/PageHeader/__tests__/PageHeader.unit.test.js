import React from 'react';
import {shallow} from 'enzyme';
import PageHeader from '../PageHeader';

describe('PageHeader', () => {
    test('renders PageHeader', () => {
        const mockColor = "green";
        const mockTitle = "Your Cart";
        const wrapper = shallow(
            <PageHeader 
                title={mockTitle} 
                color={mockColor}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});