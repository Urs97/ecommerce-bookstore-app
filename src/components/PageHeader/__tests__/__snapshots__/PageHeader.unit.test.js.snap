// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders PageHeader 1`] = `
<Fragment>
  <section
    className="page-header-nav-container"
  />
  <Element
    className="page-header-category-title page-header-bg-green"
    name="page-header"
  >
    <h1
      className="title"
    >
      Your Cart
    </h1>
  </Element>
</Fragment>
`;

exports[`PageHeader renders empty PageHeader 1`] = `
<Fragment>
  <section
    className="page-header-nav-container"
  />
  <section
    className="page-header-category-title page-header-bg-undefined"
  >
    <h1
      className="title"
    />
  </section>
</Fragment>
`;
